modules/tensor_mechanics/include/kernels/UpdatedLagrangianStressDivergence.h

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "DerivativeMaterialInterface.h"
+#include "Kernel.h"
+#include "RankFourTensor.h"
+#include "RankTwoTensor.h"
+
+/// Enforce equilibrium with an updated Lagrangian formulation
+//    This class enforces equilibrium when  used in conjunction with
+//    the corresponding strain calculator (CalculateStrainLagrangianKernel)
+//    and with either a stress calculator that provides the
+//    "cauchy_stress" and the appropriate "material_jacobian",
+//    which needs to be the derivative of the increment in Cauchy stress
+//    with respect to the increment in the spatial velocity gradient.
+//
+//    The WrapStressLagrangianKernel exists to wrap the existing MOOSE
+//    material system to provide this information, or new materials
+//    can provide it directly.
+//
+//    use_displaced_mesh must be true for large deformation kinematic
+//    The kernel enforces this with an error
+//
+class UpdatedLagrangianStressDivergence : public DerivativeMaterialInterface<Kernel>
+{
+public:
+  static InputParameters validParams();
+  UpdatedLagrangianStressDivergence(const InputParameters & parameters);
+  virtual ~UpdatedLagrangianStressDivergence(){};
```

Comment:
I think `Kernel` is already virtual, so do we need to define an empty virtual destructor here?